packages/snage/src/note/parser.ts

```diff
@@ -89,125 +68,34 @@ export const parseNote = (fields: Field[], rawNote: RawNote): TE.TaskEither<File
     );
 };
 
-const toRecord = <V>(values: Array<[string, V | undefined]>): Record<string, V> => {
-    const First = getFirstSemigroup<V>();
+const runProviders = (noteFile: string, fields: Field[]) => (
+    values: Record<string, FieldValue>
+): TE.TaskEither<string[], Record<string, FieldValue>> => {
+    const runProvider = (field: Field): TE.TaskEither<string, [string, FieldValue | undefined]> => {
+        return pipe(
+            field.provider!(noteFile),
+            TE.map((value) => [field.name, value])
+        );
+    };
+
     return pipe(
-        values,
-        A.filter((pair): pair is [string, V] => typeof pair[1] !== 'undefined'),
-        R.fromFoldable(First, A.array)
+        fields,
+        A.filter((f) => typeof f.provider !== 'undefined'),
```

Comment:
hier haben wir glaube ich ne helfer function isProvidedField oder so. Dann braucht man weiter oben den den not null assertion auch nicht (zeile 76). Wenn als parameter providedfield genutzt wird.

Reply:
Hatten wir bisher nur für die `RawField`s, aber ich hab's analog hinzugefügt.